### PR TITLE
feat: add String::ValueView

### PR DESCRIPTION
**Pull request by devsnek**
new api for faster access to string contents

### CODE REVIEW

File: src/string.rs [hunk 1]
```diff
@@ -939,3 +949,57 @@ pub extern "C" fn free_rust_external_onebyte(s: *mut char, len: usize) {
...
+/// Returns a view onto a string's contents.
+///
+/// WARNING: This does not copy the string's contents, and will therefore be
+/// invalidated if the GC can move the string while the ValueView is alive. It
+/// is therefore required that no GC or allocation can happen while there is an
+/// active ValueView. This requirement may be relaxed in the future.
+///
+/// V8 strings are either encoded as one-byte or two-bytes per character.
+#[repr(C)]
+pub struct ValueView([u8; crate::binding::RUST_v8__String__ValueView_SIZE]);
```
Comment (littledivy) [inline src/string.rs:968]: UB when GC collects the string. Maybe use lifetimes to tie `ValueView` and `ValueViewData` to `Local<'s, String>`?

File: src/string.rs [hunk 2]
```diff
@@ -939,3 +949,57 @@ pub extern "C" fn free_rust_external_onebyte(s: *mut char, len: usize) {
...
+        v.as_mut_ptr(),
+        scope.get_isolate_ptr(),
+        &*string,
+      );
+      v.assume_init()
+    }
+  }
+
+  #[inline(always)]
+  pub fn data(&self) -> ValueViewData {
```
Comment (littledivy) [inline src/string.rs:988]: This should be `unsafe`, it's possible to take a 'static reference to string contents (UB)

```rust
let view = ValueView::new(scope, local);
let data: ValueViewData<'static> = view.data(); 
```

File: src/string.rs (hunk 1 above)
Comment (littledivy) [reply src/string.rs:968]: or we can make this `unsafe` and enforce safety in op2

Comment (devsnek) [reply src/string.rs:968]: ValueView internally allocates a nogc scope so its safe.

**Review by littledivy — APPROVED** [review]
LGTM

**Pull request merged**